src/azure-cli-core/azure/cli/core/_profile.py

```diff
@@ -110,17 +111,17 @@ def find_subscriptions_on_login(self,  # pylint: disable=too-many-arguments
         allow_debug_adal_connection()
         subscriptions = []
         if interactive:
-            subscriptions = self._subscription_finder.find_through_interactive_flow(
+            subscriptions = self.subscription_finder.find_through_interactive_flow(
```

Comment:
This is an existing code defect, but since you are improving around, maybe you would not mind going one step further:
1. Let us get rid of the instance field of `subscription_finder` completely since it is only being used in this routine. 
2. For test seam injection, let us have this method take an option argument, so impacted unit test can pass in a stub. Note, your PR is already touching that test, so the code change is trivial.